### DISCUSSION TITLE
Angular 12 - help with setup

### DISCUSSION THREAD

**Discussion by StefanSCT** [elsa-workflows/elsa-core#1183] [Q&A]
Hi,

I've been looking at [this link](https://github.com/elsa-workflows/elsa-designer) for some time now trying to figure out how to implement and use this application. 
I've used the example that you have given in that link and it shows the 'Send email workflow' but there is no option for Add Activity/Import/Export/New Workflow
![image](https://user-images.githubusercontent.com/56017033/123591706-a119ac00-d7ec-11eb-82be-6c033c22b21f.png)

Am I missing something obvious? 
Did I need to download elsa-designer code and use it somehow in my new project?
I would like some guidelines on how to make Elsa Designer work?

Thank you in advance.

**Comment by sfmskywalker** [comment]
I don't know actually, I've never tried it. @RingoKam, the author of the sample, might know?

In any case, you might want to be working with the [Elsa 2 designer](https://github.com/elsa-workflows/elsa-core/tree/master/src/designer/elsa-workflows-studio) instead, since all current and future development is happening there.

**Comment by RingoKam** [comment]
Hi @StefanSCT 

You can add a new Activity by right clicking in the designer
![image](https://user-images.githubusercontent.com/10913199/123668084-c3212780-d808-11eb-85a3-d7f9b9539068.png)

For Import/Export/New Workflow, you will need to add your own UI. 
ex: "New Workflow" will be a button that pass in a new empty object to  activityDefinition/Workflow whenever clicked

As @sfmskywalker mentioned, if this is a new project and you don't NEED to use angular to integrate w/ other parts of your frontend, then you should use the Elsa 2 Designer, which is actively being developed and maintained

**Reply by StefanSCT** [reply]
Hi @RingoKam,

Thanks for clearing everything up. I would like to integrate this with an existing Angular app so I'll be sticking with this, but I'll take a look at Elsa 2 Designer as well.

Just to clear something up, are there any added features in this as opposed to Elsa Dashboard? Or is it the same app just one being able to be integrated with Angular frontend?

**Reply by sfmskywalker** [reply]
Just to clarify, both Elsa 1 and Elsa 2 components can be wrapped within Angular components following the same technique demonstrated by @RingoKam. In fact, Elsa 2 might be even easier since it offers multi-tiered components, offering you a wide range of choices of which components you would like to use.

For example, if you use the elsa-studio-root component, you get everything: a menu, editor, import/export etc.

But if you want just the workflow editor, just take elsa-workflow-definition-editor-screen.

Basically all “screen” components can be used individually.
